analyses/data-pre-release-qc/01-data-harmonization-qc.Rmd

```diff
@@ -0,0 +1,752 @@
+}
+print(paste("DNA biospecimen in cnv-controlfreec.tsv.gz missing in histolgies = ", length(missing_hist_ids)))
+```
+
+- GATK seg file (cnv-gatk.seg.gz)
+```{r}
+# check that all biospecimen IDs are in the file
+biospecimen_ids <- readr::read_tsv(file.path(data_dir, "cnv-gatk.seg.gz")) %>%
+  dplyr::pull(BS_ID) %>%
+  unique()
+hist_ids <- hist_df %>%
+  dplyr::filter(experimental_strategy %in% c("WGS", "WXS", "Targeted Sequencing"),
+                is.na(RNA_library),
+                sample_type == "Tumor",
+                cohort != "DGD",) %>%
+  dplyr::pull(Kids_First_Biospecimen_ID)
+missing_ids <- setdiff(hist_ids, biospecimen_ids)
+missing_hist_ids <- setdiff(biospecimen_ids, hist_ids)
+if (length(missing_ids) > 0) {
+  output_file <- "histologies-samples-missing-in-cnv-gatk.tsv"
+  missing_ids_df <- missing_biospecimen_table(hist_df, missing_ids, output_file)
+  missing_ids_df
+}
+print(paste("DNA biospecimen in histolgies missing in cnv-gatk.seg.gz = ", length(missing_ids)))
+if (length(missing_hist_ids) > 0) {
+  output_file <- "cnv-gatk-samples-missing-in-histologies.tsv"
+  missing_hist_ids_df <- tibble::tibble(Kids_First_Biospecimen_ID = missing_hist_ids)
+  missing_hist_ids_df %>% readr::write_tsv(file.path(results_dir, output_file))
+  missing_hist_ids_df
+}
+print(paste("DNA biospecimen in cnv-gatk.seg.gz missing in histolgies = ", length(missing_hist_ids)))
+```
+
+- SNV consensus MAF file (snv-consensus-plus-hotspots.maf.tsv.gz)
+```{r}
+# check that all biospecimen IDs are in the file
+biospecimen_ids <- data.table::fread(file.path(data_dir,
+                                               "snv-consensus-plus-hotspots.maf.tsv.gz"),
+                                     data.table = FALSE) %>%
+  dplyr::pull(Tumor_Sample_Barcode) %>%
+  unique()
+hist_ids <- hist_df %>%
+  dplyr::filter(experimental_strategy %in% c("WGS", "WXS", "Targeted Sequencing"),
+                is.na(RNA_library),
+                sample_type == "Tumor",
+                cohort != "DGD",) %>%
+  dplyr::pull(Kids_First_Biospecimen_ID)
+missing_ids <- setdiff(hist_ids, biospecimen_ids)
+missing_hist_ids <- setdiff(biospecimen_ids, hist_ids)
+if (length(missing_ids) > 0) {
+  output_file <- "histologies-samples-missing-in-snv-consensus-plus-hotspots.tsv"
+  missing_ids_df <- missing_biospecimen_table(hist_df, missing_ids, output_file)
+  missing_ids_df
+}
+print(paste("DNA biospecimen in histolgies missing in snv-consensus-plus-hotspots.maf.tsv.gz = ", length(missing_ids)))
+if (length(missing_hist_ids) > 0) {
+  output_file <- "snv-consensus-plus-hotspots-samples-missing-in-histologies.tsv"
+  missing_hist_ids_df <- tibble::tibble(Kids_First_Biospecimen_ID = missing_hist_ids)
+  missing_hist_ids_df %>% readr::write_tsv(file.path(results_dir, output_file))
+  missing_hist_ids_df
+}
+print(paste("DNA biospecimen in snv-consensus-plus-hotspots.maf.tsv.gz missing in histolgies = ", length(missing_hist_ids)))
+```
+
+- SNV DGD MAF file (snv-dgd.maf.tsv.gz)
```

Comment:
you can remove check on this bc I would expect not all samples here have mutations. Maybe a better check is column class matches the consensus SNV MAF